devdoc/Metrics.rst

```diff
@@ -0,0 +1,282 @@
+========================
+Metrics Framework Design
+========================
+
+************
+Introduction
+************
+
+This document describes the design of a metrics framework that allows HPCC Systems components to implement a
+metric collection strategy. Metrics provide the following functionality
+
+* Alerts and monitoring
+
+  An important DevOps function is to monitoring the cluster and providing alerts when
+  problems are detected. Aggregated metric values from multiple sources provide a
+  the necessary data to build a complete picture of cluster health that drives
+  monitoring and alerts.
+
+* Auto scaling
+
+  As described above, aggregated metric data is also used to dynamically respond to changing
+  cluster demands and load. Metrics provide the monitoring capability to react and take
+  action
+
+* Fault diagnosis and resource monitoring
+
+  Metrics provide historical data useful in diagnosing problems by profiling how demand and
+  usage patterns may change prior to a fault. Predictive analysis can also be applied.
+
+* Analysis of jobs/workunits and profiling
+
+  With proper instrumentation, a robust dynamic metric strategy can track workunit
```

Comment:
I think the conclusion was that this area was outside the scope of the metric framework and would be supported via stats/logging.  Possibly move the following line from the esp section here:
"Internal problems with queries should be diagnosed from deep drill down logging"

Reply:
Moved.